Content.Server/DeviceLinking/Systems/PressurePlateSystem.cs

```diff
@@ -0,0 +1,94 @@
+using Content.Server.DeviceLinking.Components;
+using Content.Shared.DeviceLinking;
+using Content.Server.Storage.Components;
+using Content.Shared.Placeable;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Physics.Events;
+
+namespace Content.Server.DeviceLinking.Systems;
+
+/// <see cref="PressurePlateComponent"/>
+public sealed class PressurePlateSystem : EntitySystem
+{
+    [Dependency] private readonly DeviceLinkSystem _signalSystem = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PressurePlateComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<PressurePlateComponent, EndCollideEvent>(OnEndCollide);
+        SubscribeLocalEvent<PressurePlateComponent, StartCollideEvent>(OnStartCollide);
+    }
+
+    private void OnInit(EntityUid uid, PressurePlateComponent component, ComponentInit args)
+    {
+        _signalSystem.EnsureSourcePorts(uid, component.PressedPort, component.ReleasedPort);
+        _appearance.SetData(uid, PressurePlateVisuals.Pressed, false);
+    }
+    private void OnStartCollide(EntityUid uid, PressurePlateComponent component, ref StartCollideEvent args)
+    {
+        UpdateState(uid, component);
+    }
+
+    private void OnEndCollide(EntityUid uid, PressurePlateComponent component, ref EndCollideEvent args)
+    {
+        UpdateState(uid, component);
+    }
+
+    private void UpdateState(EntityUid uid, PressurePlateComponent component)
+    {
+        if (!TryComp(uid, out ItemPlacerComponent? itemPlacer))
```

Comment:
style is to use `TryComp<ItemPlacerComponent>(uid, out var itemPlacer)`

also this should just use the item placer events then add or subtract mass of the item in the event args